JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -15,978 +15,589 @@
-						## assumption is met; seems like in this setting there is no
-						## sampling plan, thus the p-value is not defined. haha!
-						if (!wantsWelchs && wantsStudents) {
-							levene <- car::leveneTest(variableData, groupingData, "mean")
-
-							## arbitrary cut-offs are arbitrary
-							if (!is.na(levene[1, 3]) && levene[1, 3] < 0.05) {
-								error <- .messages('footnote', 'leveneSign')
-								foot.index <- .addFootnote(footnotes, error)
-								row.footnotes <- list(p = list(foot.index))
-
-							}
-						}
-
-						## same for all t-tests
-						p <- as.numeric(r$p.value)
-
-						ciLow <- .clean(r$conf.int[1])
-						ciUp <- .clean(r$conf.int[2])
-						lowerCIeffectSize <- .clean(as.numeric(confIntEffSize[1]))
-						upperCIeffectSize <- .clean(as.numeric(confIntEffSize[2]))
-						# this will be the results object
-						res <- list(v = variable, test = whatTest, df = df, p = p,
-												md = m, d = d, lowerCIlocationParameter = ciLow, upperCIlocationParameter = ciUp,
-												lowerCIeffectSize = lowerCIeffectSize, upperCIeffectSize = upperCIeffectSize,
-												sed = sed, .footnotes = row.footnotes)
-						res[[testStat]] <- stat
-						if (options$VovkSellkeMPR){
-							res[["VovkSellkeMPR"]] <- .VovkSellkeMPR(p)
-						}
-						res
-					})
-
-					## if there has been an error in computing the test, log it as footnote
-					if (isTryError(row)) {
-						errorMessage <- .extractErrorMessage(row)
-					}
-				}
-
-				if (!is.null(errorMessage)) {
-					## log the error in a footnote
-					index <- .addFootnote(footnotes, errorMessage)
-					row.footnotes <- list(t = list(index))
-
-					row <- list(v = variable, test = "", df = "", p = "",
-											md = "", d = "", lowerCIlocationParameter = "", upperCIlocationParameter = "",
-											lowerCIeffectSize = "", upperCIeffectSize = "",
-											sed = "", .footnotes = list(t = list(index)))
-					row[[testStat]] <- .clean(NaN)
-				}
-				## if the user only wants more than one test
-				## update the table so that it shows the "Test" and "statistic" column
-				if (sum(allTests) > 1) {
-					ttest[["schema"]] <- list(fields = fields)
-				}
-
-				ttest.rows[[rowNo]] <- row
-				rowNo <- rowNo + 1
-			}
-		}
-
-		if (effSize == "glass") {
-			sdMessage <- paste0("Glass' delta uses the standard deviation of group ", names(ns[2]),
-										 " of variable ", options$groupingVariable, ".")
-			.addFootnote(footnotes, symbol = "<em>Note.</em>", text = sdMessage)
-		}
-
-
-		ttest[["footnotes"]] <- as.list(footnotes)
-	}
-
-	ttest[["data"]] <- ttest.rows
-	ttest
+.ttestIndependentMainTable <- function(jaspResults, dataset, options, ready, type) {
+  if (!is.null(jaspResults[["ttest"]])) 
+    return()
+  optionsList <- .ttestOptionsList(options, type)
+  # Create table
+  ttest <- createJaspTable(title = "Independent Samples T-Test")
+  dependList <- c("effectSize", "effSizeConfidenceIntervalCheckbox", "variables",
```

Comment:
This misses a dependency on the actual value of the effect size confidence interval (e.g., `95`)